### PR TITLE
Docs: Update documentation with new SAML features

### PR DESCRIPTION
**Pull request by gotjosh**
**What this PR does / why we need it**:

- Updates `sample.ini` and `defaults.ini` with the mapping configuration options
- Updates the SAML documentation with the assertion mapping configuration options.

### CODE REVIEW

**Review by alexanderzobnin — APPROVED** [review]
LGTM!

File: docs/sources/auth/saml.md [hunk 1]
```diff
@@ -91,14 +100,18 @@ metadata_valid_duration = 48h
...
-| `max_issue_delay`         | No       | Duration, since the IdP issued a response and the SP is allowed to process it  | `90s`   |
-| `metadata_valid_duration` | No       | Duration, for how long the SP's metadata should be valid                       | `48h`   |
+| Setting                     | Required | Description                                                                                        | Default                                                     |
+|-----------------------------|----------|----------------------------------------------------------------------------------------------------|-------------------------------------------------------------|
+| `eanbled`                   | No       | Whenever SAML authentication is allowed                                                            | `false`                                                     |
+| `certificate                | _path`   | Yes                                                                                                | Base64-encoded string or Path for the SP X.509 certificate  |
```
Comment (xlson) [inline docs/sources/auth/saml.md]: We have a rogue `|` here that makes _path end up in the Required column and everything else pushed a column as well.

Perhaps try without the `|` (which could be considered a bit technical) and go with something like this: `certificate` or `certificate_path`

File: docs/sources/auth/saml.md [hunk 2]
```diff
@@ -91,14 +100,18 @@ metadata_valid_duration = 48h
...
-| `private_key|_path`       | Yes      | Base64-encoded string or Path for the SP private key                           |         |
-| `idp_metadata|_path`      | Yes      | Base64-encoded string or Path for the IdP SAML metadata XML                    |         |
-| `max_issue_delay`         | No       | Duration, since the IdP issued a response and the SP is allowed to process it  | `90s`   |
-| `metadata_valid_duration` | No       | Duration, for how long the SP's metadata should be valid                       | `48h`   |
+| Setting                     | Required | Description                                                                                        | Default                                                     |
+|-----------------------------|----------|----------------------------------------------------------------------------------------------------|-------------------------------------------------------------|
+| `eanbled`                   | No       | Whenever SAML authentication is allowed                                                            | `false`                                                     |
```
Comment (xlson) [inline docs/sources/auth/saml.md]:
```suggestion
| `enabled`                   | No       | Whenever SAML authentication is allowed                                                            | `false`                                                     |
```

File: docs/sources/auth/saml.md (hunk 1 above)
Comment (xlson) [reply docs/sources/auth/saml.md]: Same goes for the other similar cases.

**Review by xlson — CHANGES_REQUESTED** [review]
Great work on the docs, very comprehensive.

The SAML docs page should contain a blurb about this being an Enterprise feature, just as at the top of https://grafana.com/docs/auth/enhanced_ldap/

SAML should be mentioned on the Grafana Enterprise page with a link back to the SAML docs.

**Review by xlson — APPROVED** [review]
Looks great!

**Pull request merged**